cmd/kubeadm/app/phases/kubelet/flags.go

```diff
@@ -57,8 +56,10 @@ func WriteKubeletDynamicEnvFile(cfg *kubeadmapi.ClusterConfiguration, nodeReg *k
 		pauseImage:               images.GetPauseImage(cfg),
 		registerTaintsUsingFlags: registerTaintsUsingFlags,
 		execer:                   utilsexec.New(),
-		pidOfFunc:                procfs.PidOf,
-		defaultHostname:          hostName,
+		isProcessRunningFunc: func(name string) bool {
+			return utilsexec.New().Command("pidof", name).Run() == nil
```

Comment:
1) is `pidof` available on all of our top distros that we target? my guess is yes?
(asked on a certain #linux slack channel too)
2) the original logic is doing some manual parsing:
https://github.com/kubernetes/kubernetes/blob/master/pkg/util/procfs/procfs_linux.go#L96-L165
maybe only for the sake of not calling `pidof`.

here is a reference source for pidof.c:
https://github.com/thlorenz/procps/blob/master/deps/procps/pidof.c
and it seems to be doing something similar.

i'm +1 on the change if we are OK with 1).

Reply:
collected a bunch of feedback and it seems the binary is quite available.
i suggest that we go for it...

/lgtm
/assign @rosti 
for approve.

Reply:
it's unfortunate to depend on more shelling out because we couldn't get a small shared go package for this :( the procfs package seems like a reasonable candidate for k8s.io/utils

Reply:
@bart0sh it might be safer to not fail if the `pidof` binary is not found.
so first we need to check if the command is OK and only then determine if systemd-resolved is running.
/lgtm cancel

Reply:
> it's unfortunate to depend on more shelling out because we couldn't get a small shared go package for this :( the procfs package seems like a reasonable candidate for k8s.io/utils

we didn't try for this package, but thus far everything is rejected so.
maybe the safer option is to duplicate the logic under kubeadm and not shellout.

leaving this to @rosti at this point.
/cc @yastij 
too.

Reply:
@thockin  - would this fit into the kernel package we discussed under https://github.com/kubernetes/utils/pull/99#discussion_r303085545 ?

Reply:
> it's unfortunate to depend on more shelling out because we couldn't get a small shared go package for this :( the procfs package seems like a reasonable candidate for k8s.io/utils

There is a bright side in this too. pidof works not just on linux, but on pretty much any unix. Considering amount of shelling out in the kubeadm codebase it's not a big deal in my opinion.

Anyway, I've submitted PR to k/utils as you've suggested: https://github.com/kubernetes/utils/pull/104. Feel free to comment there.

Reply:
Sorry for coming late to the party. `systemd-resolved` is a standard service name. Hence it should be able to query it with something like (error checks omitted for the sake of simplicity):
```golang
initSystem, _ := initsystem.GetInitSystem()
if initSystem.ServiceIsActive("systemd-resolved") {
  kubeletFlags["resolv-conf"] = "/run/systemd/resolve/resolv.conf"
}
```

Then we can get rid of `pidof` as this is the only use case for it.

Reply:
@rosti thank you! That's much better than running pidof for sure!

Updated the PR, please review.